File: w09/d01/katherine_santiago/mail_item/lib/package.rb
class Package < Mailitem

  def initialize
    @content = []
  end

  def content
    @content
  end

  def is_fragile?
    true
  end

  def is_broken?
    if (is_fragile? && shake)
      true
    end
  end

  def shake
    true
  end

end

File: w09/d03/jimmy_davis/mail_objects_app/app/assets/javascripts/package.js
function Package() {
  this.fragile = false;
  this.content = [];
  this.broken = false;
}

Package.prototype.addContent = function(content) {
  this.content.push(content);
};

Package.prototype.isFragile = function(boo) {
  if( typeof boo == "boolean" ) {
    this.fragile = boo;
  }
  else {
    throw "Must be true or false";
  }
};

Package.prototype.shake = function() {
  if( this.fragile === true ) {
    this.broken = true;
  }
};

Package.prototype.isBroken = function() {
 return this.broken;
};
